internal/server/steady_state.go

```diff
@@ -18,53 +18,59 @@ const (
 // SteadyState is the handler for the `/v1/steady-state` endpoint.
 func (a *Airlock) SteadyState() http.Handler {
 	handler := func(w http.ResponseWriter, req *http.Request) {
-		logrus.Debug("got steady-state report")
-		if a == nil {
-			http.Error(w, errNilAirlockServer.Error(), 500)
-			return
-		}
-
-		nodeIdentity, err := validateIdentity(req)
+		code, err := a.preRebootHandler(w, req)
 		if err != nil {
-			logrus.Errorln("failed to validate client identity: ", err)
-			http.Error(w, err.Error(), 400)
-			return
+			http.Error(w, err.Error(), code)
 		}
-		logrus.WithFields(logrus.Fields{
-			"group": nodeIdentity.Group,
-			"uuid":  nodeIdentity.UUID,
-		}).Debug("processing client steady-state report")
+	}
 
-		slots, ok := a.LockGroups[nodeIdentity.Group]
-		if !ok {
-			err := fmt.Errorf("unknown group %q", nodeIdentity.Group)
-			logrus.Errorln("unable to satisfy client request: ", err)
-			http.Error(w, err.Error(), 400)
-			return
-		}
+	return http.HandlerFunc(handler)
+}
 
-		ctx, cancel := context.WithTimeout(context.Background(), a.EtcdTxnTimeout)
-		defer cancel()
-		lockManager, err := lock.NewManager(ctx, a.EtcdEndpoints, nodeIdentity.Group, slots)
-		if err != nil {
-			logrus.Errorln("failed to initialize semaphore manager: ", err)
-			http.Error(w, err.Error(), 500)
-			return
-		}
-		defer lockManager.Close()
+// steadyStateHandler contains logic to handle steady-state.
+func (a *Airlock) steadyStateHandler(w http.ResponseWriter, req *http.Request) (int, error) {
+	logrus.Debug("got steady-state report")
+	if a == nil {
+		http.Error(w, errNilAirlockServer.Error(), 500)
```

Comment:
It feels like this should be a `logrus.Errorln` or dropped.

Reply:
Indeed, good catch, I missed it when moving stuff around!